numpy/core/src/npymath/ieee754.c.src

```diff
@@ -789,30 +789,72 @@ int npy_clear_floatstatus_barrier(char *param)
 /* General GCC code, should work on most platforms */
 #  include <fenv.h>
 
+// musl-libc defines all the fenv functions unconditionally (fesetexcept,
+// feclearexcept, etc) but on unsupported platforms they are no-ops:
+// https://git.musl-libc.org/cgit/musl/tree/src/fenv/fenv.c
+// However, the platform is expected to only define flags like
+// FE_OVERFLOW if they are supported. I haven't found an explanation
+// of the design decision, but it seems to be aimed at fine-grained
+// feature detection. musl-libc-test has code that uses fenv and wants
+// it to silently decay to a no-op on platforms missing support. I
+// copied their implementation of this behavior:
+// http://nsz.repo.hu/git/?p=libc-test;a=blob;f=src/common/mtest.h;h=706c1ba23ea8989b17a2f72ed1a919e187c06b6a;hb=HEAD#l30
+#undef INEXACT
+#undef INVALID
+#undef DIVBYZERO
+#undef UNDERFLOW
+#undef OVERFLOW
+#ifdef FE_INEXACT
+#define INEXACT FE_INEXACT
+#else
+#define INEXACT 0
+#endif
+#ifdef FE_INVALID
+#define INVALID FE_INVALID
+#else
+#define INVALID 0
+#endif
+#ifdef FE_DIVBYZERO
+#define DIVBYZERO FE_DIVBYZERO
+#else
+#define DIVBYZERO 0
+#endif
+#ifdef FE_UNDERFLOW
+#define UNDERFLOW FE_UNDERFLOW
+#else
+#define UNDERFLOW 0
+#endif
+#ifdef FE_OVERFLOW
+#define OVERFLOW FE_OVERFLOW
+#else
+#define OVERFLOW 0
+#endif
```

Comment:
This is a local C-file, any reason not to shorten this to?
```C
#ifndef FE_INEXACT
    #define FE_INEXACT 0
#endif
```
I might also prefer shorting the comment a bit:
```C
/*
 * According to the C99 standard FE_INEXACT, etc. may
 * not be provided when not supported.
 * In such cases NumPy will not report these correctly,
 * but we should still allow compiling making all FP error
 * handling effectively no-ops.
 */
```
(Maybe continue with a reference to musl?)

Anyway, super nitpicky, so am happy to just fix it up myself or keep it if you think it is no improvement.

Reply:
IMO the two musl links should stay since they are probably helpful. Other than that your reworded comment seems good to me.